test_runner/fixtures/zenith_fixtures.py

```diff
@@ -0,0 +1,326 @@
+
+    If the environment variable TEST_SHARED_FIXTURES is set, then all
+    tests will share the same database. State, logs, etc. will be
+    stored in a directory called "shared".
+
+    """
+    if os.environ.get('TEST_SHARED_FIXTURES') is None:
+        scope = 'function'
+    else:
+        scope = 'session'
+    return pytest.fixture(func, scope=scope)
+
+
+@pytest.fixture(autouse=True, scope='session')
+def safety_check():
+    """ Ensure that no unwanted daemons are running before we start testing. """
+    cmd = ['pgrep', '-c', 'pageserver|postgres']
+    result = subprocess.run(cmd)
+    if result.returncode == 0:
+        # returncode of 0 means it found something.
+        # This is bad; we don't want any of those processes polluting the
+        # result of the test.
+        raise Exception('found interfering processes running')
+
+
+class ZenithCli:
+    """ An object representing the CLI binary named "zenith".
+
+    We also store an environment that will tell the CLI to operate
+    on a particular ZENITH_REPO_DIR.
+    """
+
+    def __init__(self, binpath, repo_dir, pg_distrib_dir):
+        assert os.path.isdir(binpath)
+        self.binpath = binpath
+        self.bin_zenith = os.path.join(binpath, 'zenith')
+        self.env = os.environ.copy()
+        self.env['ZENITH_REPO_DIR'] = repo_dir
+        self.env['POSTGRES_BIN'] = pg_distrib_dir
+
+    def run(self, arguments):
+        """ Run "zenith" with the specified arguments.
+
+        arguments must be in list form, e.g. ['pg', 'create']
+        """
+        assert type(arguments) == list
+        args = [self.bin_zenith] + arguments
+        print('Running command "{}"'.format(' '.join(args)))
+        subprocess.run(args, env=self.env, check=True)
+
+    def run_init(self):
+        """ Run the "zenith init <dir>" command. """
+        self.run(['init'])
+
+
+@zenfixture
+def zenith_cli(zenith_binpath, repo_dir, pg_distrib_dir):
+    return ZenithCli(zenith_binpath, repo_dir, pg_distrib_dir)
+
+
+class ZenithPageserver:
+    """ An object representing a running pageserver. """
+
+    def __init__(self, zenith_cli):
+        self.zenith_cli = zenith_cli
+        self.running = False
+
+    def start(self):
+        self.zenith_cli.run(['pageserver', 'start'])
+        self.running = True
+
+    def stop(self):
+        # FIXME: this is a todo!() in the zenith cli code
+        if self.running:
+            try:
+                subprocess.run(['killall', 'pageserver'])
+            except FileNotFoundError:
+                print(
+                    'WARNING: Failed to terminate pageserver, "killall" not found', file=sys.stderr)
+
+
+@zenfixture
+def pageserver(zenith_cli):
+    ps = ZenithPageserver(zenith_cli)
+    yield ps
+    # After the yield comes any cleanup code we need.
+    print('Starting pageserver cleanup')
+    ps.stop()
+
+
+class Postgres:
+    """ An object representing a running pageserver. """
```

Comment:
s/pageserver/postgres

Reply:
Will fix. Sorry, I forgot to do this change before merging.